### DISCUSSION TITLE
Development workflow (CI/CD)

### DISCUSSION THREAD

**Discussion by samuelcastro** [hasura/graphql-engine#3939] [Question]
Is there any guide around how to structure a good CI/CD workflow or/and how it should be implemented on CircleCI or GitHub Actions for example.

Another thing (might be a dumb question), can we track changes through git? Let's say I changed schemas in my database or created events on Hasura, I want to be able to track those changes. I'm not sure yet whether this is something trackable.

Thanks.

**Comment by samuelcastro** [comment]
So for the second part of my question I just learned that I can track any changes through migrations (https://docs.hasura.io/1.0/graphql/manual/migrations/index.html).

**Comment by marionschleifer** [comment]
@samuelcastro exactly. You can track changes through [migrations & metadata](https://hasura.io/docs/1.0/graphql/manual/migrations/index.html).

Regarding your CI/CD question, please refer to [this issue](https://github.com/hasura/graphql-engine/issues/3649). Hopefully it will help you set up CI/CD with GitHub Actions 🙂
